contrib/ansible/library/skydive_edge.py

```diff
@@ -0,0 +1,253 @@
+      node2: "{{ port1_result.UUID }}"
+      relation_type: ownership
+  - name: link tor and port 2
+    skydive_edge:
+      node1: "{{ tor_result.UUID }}"
+      node2: "{{ port2_result.UUID }}"
+      relation_type: ownership
+  - name: link port 1 and eth0
+    skydive_edge:
+      node1: "{{ port1_result.UUID }}"
+      node2: G.V().Has('Name', 'tun0')
+      relation_type: layer2
+'''
+
+RETURN = '''
+original_message:
+    description: The original name param that was passed in
+    type: str
+message:
+    description: The output message that the sample module generates
+'''
+
+import uuid
+
+from ansible.module_utils.basic import AnsibleModule
+
+from skydive.graph import Node, Edge
+from skydive.rest.client import RESTClient
+
+from skydive.websocket.client import NodeAddedMsgType, EdgeAddedMsgType
+from skydive.websocket.client import WSClient, WSClientDefaultProtocol, WSMessage
+
+
+class EdgeInjectProtocol(WSClientDefaultProtocol):
+
+    def onOpen(self):
+        module = self.factory.kwargs["module"]
+        params = self.factory.kwargs["params"]
+        result = self.factory.kwargs["result"]
+
+        node1 = self.factory.kwargs["node1"]
+        node2 = self.factory.kwargs["node2"]
+
+        if module.check_mode:
+            self.stop()
+            return
+
+        try:
+            host = params["host"]
+
+            metadata = params["metadata"]
+            metadata["RelationType"] = params["relation_type"]
+
+            uid = uuid.uuid5(uuid.NAMESPACE_OID, "%s:%s:%s" %
+                            (node1, node2, params["relation_type"]))
+
+            edge = Edge(str(uid), host,
+                        node1, node2, metadata=metadata)
+
+            msg = WSMessage("Graph", EdgeAddedMsgType, edge)
+            self.sendWSMessage(msg)
+
+            result["UUID"] = str(uid)
+        except Exception as e:
+            module.fail_json(
+                msg='Error during topology update %s' % e, **result)
+        finally:
+            self.stop()
+
+
+def get_node_id(module, node_selector):
+    scheme = "http"
+    if module.params["ssl"]:
+        scheme = "https"
+
+    if node_selector.startswith("G.") or node_selector.startswith("g."):
+        restclient = RESTClient(module.params["analyzer"], scheme=scheme,
+                                insecure=module.params["insecure"],
+                                username=module.params["username"],
+                                password=module.params["password"])
+        nodes = restclient.lookup_nodes(node_selector)
+        if len(nodes) == 0:
+            raise Exception("Node not found: %s" % node_selector)
+        elif len(nodes) > 1:
+            raise Exception("Node selection should return only one node: %s" % node_selector)
+       
+        return str(nodes[0].id)
+
+    return node_selector
+
+
+def run_module():
+    module_args = dict(
+        analyzer=dict(type='str', default="localhost:8082"),
+        ssl=dict(type=bool, default=False),
```

Comment:
`type=bool` need to be replaced by `type='bool'`
and
`type=str` by `type='str'`